zebra-consensus/src/block/subsidy/founders_reward.rs

```diff
@@ -26,14 +40,100 @@ pub fn founders_reward(height: Height, network: Network) -> Result<Amount<NonNeg
     }
 }
 
+/// Function `FounderAddressChangeInterval` as specified in [protocol specification §7.8][7.8]
+///
+/// [7.8]: https://zips.z.cash/protocol/canopy.pdf#foundersreward
+pub fn founders_address_change_interval() -> Height {
+    let interval = div_ceil(
+        SLOW_START_SHIFT.0 + PRE_BLOSSOM_HALVING_INTERVAL.0,
+        FOUNDERS_ADDRESS_COUNT,
+    );
+    Height(interval)
+}
+
+/// Get the founders reward t-address for the specified block height as described in [protocol specification §7.8][7.8]
+///
+/// [7.8]: https://zips.z.cash/protocol/canopy.pdf#foundersreward
+pub fn founders_reward_address(height: Height, network: Network) -> Result<Address, Error> {
+    let blossom_height = Blossom
+        .activation_height(network)
+        .expect("blossom activation height should be available");
+
+    if !founders_reward_active(height, network) {
+        panic!("no address returned after canopy");
```

Comment:
Panics and errors should explain the problem, and why it happened

```suggestion
        panic!("founders reward address lookup on invalid block: block is after canopy activation");
```